### ISSUE TITLE
question for NCCL write data size

### ISSUE THREAD

**Issue by gabbychen** [NVIDIA/nccl-tests#266]
Hi

        When I utilize NCCL for data sending/receiving
        I found the write data size to memory is doubled than the received data size
        Is there any reason for the duplicated data write?

**Comment by kiskra-nvidia** [comment]
Memory buffers not being registered perhaps? https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/usage/bufferreg.html

**Comment by gabbychen** [comment]
Hi, I utilized nccl_test for profiling.
What I found is the write data size in memory is doubled than the received data size I set in nccl_test
(e.g. I set the total data size as 1 GB for AllGather profiling with 4 GPU, the write data size is 1.5 GB for in place (2 x 0.75 GB (received data) and 1.75 GB for out of place,
when I utilized the same data size with 2 GPU, the write data size is 1 GB (2 x 0.5 GB (received data size) for inplace and 1.5GB for out-of-place)
I thought it's caused by internal mechanism.

I wonder if it's caused by extra buffer on communication or some other reason?
